tests: Introduce an e2e testing framework for Amazon Q (#4254)
aws/aws-toolkit-vscode: docs/TESTPLAN.md

-   E2E Tests: **slow** tests
    -   Live in `src/testE2E`
    -   These tests are heavier than Integration tests.

>     -   Some E2E tests have a more complicated architecture. The document explaining that architecture can be found in [E2E_TESTING](./E2E_TESTING_ARCHITECTURE.md)

Reviewer: ```suggestion
    -   Some E2E tests have a more complicated architecture, described in [E2E_TESTING](./E2E_TESTING_ARCHITECTURE.md)
```
